plugins/device/transaction.go

```diff
@@ -59,3 +62,56 @@ func txnPrepareDevice(c transaction.TxnCtx) error {
 	}
 	return nil
 }
+
+func txnDeleteDevice(c transaction.TxnCtx) error {
+	var err error
+	var peerID string
+
+	if err = c.Get("peerid", &peerID); err != nil {
+		return err
+	}
+
+	var deviceName string
+	if err = c.Get("device", &deviceName); err != nil {
+		return err
+	}
+
+	devices, err := deviceutils.GetDevices(peerID)
+	if err != nil {
+		return err
+	}
+
+	if len(devices) == 0 {
+		return errors.ErrDeviceNotFound
+	}
+
+	vgName := ""
+	for _, device := range devices {
+		if device.Device == deviceName {
+			vgName = device.VgName()
+		}
+	}
+
+	if vgName == "" {
+		return ErrDeviceNameNotFound
+	}
+
+	// Remove VG
+	if err = lvmutils.RemoveVG(vgName); err != nil {
```

Comment:
what happens when the VG is already deleted in the last execution and something failed in PV deletion below? 
before deleting the resource don't we need to check if the resource is available or not?

Reply:
@Madhu-1 Can you elaborate more on `whether resource is available or not`. Do you mean whether the device exist or in use or not?

Reply:
@rishubhjain during the device removal if the VG removal was successful and it got failed in PV remove stage, and if the user again tries to delete the device, it will fail with  VG  not found error right, how are we solving this problem?
dont we need to handle VG not found scenario and PV not found scenario?

Reply:
@aravindavk Can you help me with this? the basic way would be to check pvs and vgs before removing to avoid issues. But it's not a clean solution, can you suggest a cleaner solution to handle this scenerio?

Reply:
try to delete VG and continue if it is not exists error. But if Vg is not found make sure no Vg exists before removing PV.(If user manually deleted Vg and Created Vg with different name then gd2 may end up deleting, If PV fails when Vg exists then that is also Ok)

Reply:
Also check PV not exists. If Pv not exists then remove only from store and return success

Reply:
This is only possible if user manually deletes PV right?